### DISCUSSION TITLE
sufficient mDBC boundary layer criteria

### DISCUSSION THREAD

**Discussion by wwwjones** [DualSPHysics/DualSPHysics#214] [DUALSPHYSICS_V5.2]
Hi all, I was wondering if anybody knew a "good enough" criteria for determining how many boundary layers are necessary in mDBC use cases. 

According to the mDBC guide, layers>=2h/dp, though it seems this criteria is much too strict. If we take the mDBC example 09_FloatingDuck as an example, coefh is defined as 1.8, which would mean, according to our criteria, that we would need, at least 6.24 layers, though the example only constructs 3 layers, which should results in a coefh of at most 0.867...

I have tried toying around with stl files to define the boundaries, and if I keep coefh at 1.8 and scale dp so that my walls are 3 layers thick, the fluid simply passes through the boundary. Now, a coefh of 1.8 means we have a an hdp (which is coefh*sqrt(3)) of 3.12, which is clearly higher than our allegedly necessary 1.5 for 3 layers, so this makes sense. However, I have done some testing and found that an hdp of 2 still yields results where boundaries act like boundaries.

Scaling dp so that the same geometry fits the necessary 6.24 layers works, but seems overkill and takes far too long.
So my question is, what works best for you, is there any theoretically defined middle ground that works well enough, and what is the suggested way to pick a dp and coefh? 

Thank you!

**Comment by EnglishA1** [comment]
Hi, 

Basically you need to have enough boundary layers so that the smoothing kernel of a fluid particles close to the boundary is complete. The kernle has a radius of 2h and so you want to make sure the last boundary layer is at least 2h from the fluid particles. This is why we suggest layer>=2h/dp. I personally always use the hdp definition for my kernel as is is easier to calculate boundary thickness than coefh, and tend to use a hdp value of between 1.3 and 2, giving me 3 or 4 boundary layers, I also always round up so would use 3 layes fro hdp=1.3 even if 2.6 layers would theoretically be enough.

Hope this helps

**Comment by alexbexe** [comment]
The example in examples\mdbc\09_FloatingDuck will apply mDBC only for the STL duck.
For the duck we follow what you calculated of minimum of 6 layers of boundary particles inside the STL (STL is filled with particles).

The case builds 3 layers for the tank, but we are not applying mDBC in the tank... only in the floating duck.

To sum up, all cases use enough layers filling 2h for the boundaries where we will apply mDBC.

Regards
Alex
